File: freight-triage-polyglot-ledger-recovery/environment/native/src/stats/sum_milli.cpp
#include "freight/stats.h"

#include <algorithm>

namespace freight {

// sum_milli kernel.
long long stat_sum_milli(const std::vector<long long>& series) {
  long long total = 0;
  for (size_t i = 0; i < series.size(); ++i) {
    total += series[i];
  }
  return total * 1000;
}

}  // namespace freight

File: freight-triage-polyglot-ledger-recovery/environment/intake/src/com/freight/stats/SumMilli.java
package com.freight.stats;

/** sum_milli kernel. */
public final class SumMilli implements StatKernel {

    @Override
    public String name() {
        return "sum_milli";
    }

    @Override
    public long apply(long[] series) {
        long total = 0L;
        for (int i = 0; i < series.length; i++) {
            total += series[i];
        }
        return total * 1000L;
    }
}
